packages/snjs/test/lib/model_tests/mapping.test.js

```diff
@@ -5,29 +5,20 @@ import * as Factory from '../../factory';
 describe('model manager mapping', () => {
   const BASE_ITEM_COUNT = 2; /** Default items key, user preferences */
 
-  let expectedItemCount;
-  let application;
-
-  beforeEach(async function () {
-    expectedItemCount = BASE_ITEM_COUNT;
-    application = await Factory.createInitAppWithRandNamespace();
-  });
-
-  afterEach(async function () {
-    application.deinit();
```

Comment:
Where are the applications in each test being de-inited now that it's not happening automatically? It needs to happen at the end of every test I believe.

You can maybe call `context.cleanup()` after each test.

Reply:
We don't need to deinit application instances after each test. This is done automatically by the test environment, which is reset after each test

Reply:
Really? Can you link to docs about this? Not sure what "reset" means though? And how does this differ from Mocha behavior?

Reply:
Basically, a new test environment is created for each test file. After all tests within a test file have finished, everything is removed/deinitialized/closed.

See [setupFiles](https://jestjs.io/docs/configuration#setupfiles-array) and [setupFilesAfterEnv](https://jestjs.io/docs/configuration#setupfilesafterenv-array).

> Since every test runs in its own environment, these scripts will be executed in the testing environment immediately before executing the test code itself.

Reply:
If it's per file then that's not sufficient. That means that tests within a file could have multiple applications writing to the same storage/syncing/living simultaneously.